doc/usecases/core/UC_keyset_delete.md

```diff
@@ -0,0 +1,28 @@
+# Use Case: Delete `Key` from `KeySet`
+
+## Summary
+
+- **Title:** Delete `Key` from `KeySet`
+- **Scope:** `libelektra-core`
+- **Level:** Developer Goal
+- **Actors:** Core, Caller
+- **Brief:** Caller removes `Key` from existing `KeySet`
+
+## Scenarios
+
+- **Precondition:**
+  - [`KeySet` has been created](UC_keyset_create.md)
+  - [`Key` has been inserted into `KeySet`](UC_keyset_insert.md)
+- **Main success scenario:**
+  - Caller [looks up `Key` in `KeySet`](UC_keyset_lookup_basic.md) to get index
+  - Caller requests to remove `Key` by index from `KeySet`
+  - Core removes given index from `KeySet`
+- **Alternative scenario:**
+  - Caller [looks up hierarchy in `KeySet`](UC_keyset_lookup_prefix.md) to get index range
+  - Caller requests to remove `Key`s by index range from `KeySet`
+  - Core removes given index range from `KeySet`
+- **Error scenario:** -
+- **Postcondition:**
+  - The deleted `Key`(s) MUST NOT be part of `KeySet` anymore.
+    A lookup for that/those `Key`s MUST NOT return any results.
```

Comment:
We should also be clear in the use cases that we really have an array-like access.

```suggestion
    A lookup for that/those `Key`s MUST NOT return any results.
    Indexes of the `Key`s after the removed key in the `KeySet` change by one.
```

Reply:
No, IMO that's not necessary. The lookup use case already states that indices are only valid until a key is added/removed. That's enough IMO. That there is actually just a single array behind and the index changes by one, is an implementation detail.

I've been very careful in how I describe the data structure of a `KeySet`. If we go into too much detail, the use case already precludes the decision for how a `KeySet` is implemented. For example, I'm sure there is a way to build `KeySet` as a Radix tree or a Judy array.